File: Native/Test/Fabric/case-164.js
F = require('Fabric').createClient();

operator = F.DG.createOperator("op");
operator.setEntryFunctionName("entry");
operator.setSourceCode("\
operator entry( io Size foo ) {\n\
  report 'foo = ' + foo;\n\
}\n\
");
if (operator.getDiagnostics().length > 0 )
  console.log(operator.getDiagnostics());

binding = F.DependencyGraph.createBinding();
binding.setOperator(operator);
binding.setParameterLayout(['self.bar','self.foo']);

node = F.DG.createNode("node");
node.addMember('bar', 'Size');
node.setData('bar', 0, 42);
node.bindings.append(binding);
if (node.getErrors().length > 0)
  console.log(node.getErrors());
node.evaluate();

F.flush();

File: Native/Test/Python/case-164.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric
F = fabric.createClient()

operator = F.DG.createOperator("op")
operator.setEntryFunctionName("entry")
operator.setSourceCode("\
operator entry( io Size foo ) {\n\
  report 'foo = ' + foo;\n\
}\n\
")

if len( operator.getDiagnostics() ) > 0:
  print(operator.getDiagnostics())

binding = F.DependencyGraph.createBinding()
binding.setOperator(operator)
binding.setParameterLayout(['self.bar','self.foo'])

node = F.DG.createNode("node")
node.addMember('bar', 'Size')
node.setData('bar', 0, 42)
node.bindings.append(binding)
if len( node.getErrors() ) > 0:
  print(node.getErrors())
try:
  node.evaluate()
except Exception as e:
  print(e)

F.close()
